common/src/sql/dbinit.sql

```diff
@@ -709,12 +705,20 @@ CREATE TABLE omicron.public.saml_identity_provider (
     group_attribute_name TEXT
 );
 
-CREATE INDEX ON omicron.public.saml_identity_provider (
-    id,
-    silo_id
+CREATE UNIQUE INDEX ON omicron.public.saml_identity_provider (
+    silo_id,
+    id
+) WHERE
+    time_deleted IS NULL;
+
+CREATE UNIQUE INDEX ON omicron.public.saml_identity_provider (
+    silo_id,
+    name
```

Comment:
I added this to ensure that SamlIdentityProvider names are unique within a Silo.